### PR TITLE
Learner search bar + search page updates

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

**Give the search bar a rounded solid border, and reduce font size on App Bar**

1. Also, re-focus on input after clearing with X button or "ESC"

| Before | After |
|--------|-------|
| Screen Shot 2019-07-03 at 12 37 20 PM|Screen Shot 2019-07-03 at 12 37 39 PM |

**Improve behavior when on Search Results Page**

1. Focus on search input automatically
1. Automatically select the search term if there are no results.
1. Hide filters when there are no results
1. Expand the channel select menu to fit the longest channel names
1. Clear out the search term in vuex when leaving the page (without clearing it, the search field would still be filled with previous search term after exiting search results)

| Before | After |
|--------|-------|
| Screen Shot 2019-07-03 at 12 38 12 PM    | Screen Shot 2019-07-03 at 12 37 58 PM      |

### Reviewer guidance

1. Test out  the learner search bar on different browsers
1. Test it out only using keyboard
…

### References

…

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by indirectlylit** [comment]
nice updates!!

File: kolibri/plugins/learn/assets/src/views/SearchPage.vue [hunk 1]
```diff
@@ -74,6 +74,21 @@
     computed: {
       ...mapState('search', ['contents', 'searchTerm', 'total_results']),
     },
+    beforeDestroy() {
+      // TODO do this clean up in a beforeRouteLeave once SearchPage is rendered in router-link
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/views/SearchPage.vue:78]: is this essentially the same as what's proposed in https://github.com/learningequality/kolibri/issues/5719 ?

Comment (jonboiser) [reply kolibri/plugins/learn/assets/src/views/SearchPage.vue:78]: Yeah, implementing #5719 will give us access to those vue router hooks in this page

Comment (jonboiser) [reply kolibri/plugins/learn/assets/src/views/SearchPage.vue:78]: There’s also some other similar vuex module cleanup code that is run in a central router hook that could be potentially moved to a in-component  hook in the same way

**Review by indirectlylit — APPROVED** [review]

**Pull request merged**